SqlInstall.ps1

```diff
@@ -0,0 +1,433 @@
+
+    (Get-Content -Path $configini).Replace('SQLUSERDBDIR="E:\Program Files\Microsoft SQL Server\MSSQL12.AXIANSDB01\MSSQL\Data"', 'SQLUSERDBDIR="' + $DataVolume + ':\Program Files\Microsoft SQL Server\MSSQL12.AXIANSDB01\MSSQL\Data"') | Out-File $configini
+
+    (Get-Content -Path$configini).Replace('SQLTEMPDBDIR="E:\Program Files\Microsoft SQL Server\MSSQL12.AXIANSDB01\MSSQL\Data"', 'SQLTEMPDBDIR="' + $TempVolume + ':\Program Files\Microsoft SQL Server\MSSQL12.AXIANSDB01\MSSQL\Data"') | Out-File $configini
+
+    (Get-Content -Path $configini).Replace('SQLUSERDBLOGDIR="E:\Program Files\Microsoft SQL Server\MSSQL12.AXIANSDB01\MSSQL\Log"', 'SQLUSERDBLOGDIR="' + $LogVolume + ':\Program Files\Microsoft SQL Server\MSSQL12.AXIANSDB01\MSSQL\Log"') | Out-File $configini
+
+    (Get-Content -Path $configini).Replace('SQLSYSADMINACCOUNTS="WIN-NAJQHOBU8QD\Administrator"', 'SQLSYSADMINACCOUNTS="' + $SqlServerAccount)| Out-File $configini
+
+    If ($SaveFile -eq "Yes") {
+        $SaveFileLocation = Read-Host "Please enter your preferred directory for saving a copy of the configuration file: "
+        Copy-Item $configini -Destination $SaveFileLocation
+    }
+
+    IF ($Authentication -eq "Mixed Mode") {
+        $SAPassW = [PsCredential](Get-Credential -UserName "SA"  -Message "Please Enter the SA Password.")
+
+        & $SetupFile /ConfigurationFile=$configini /Q /IACCEPTSQLSERVERLICENSETERMS /SAPWD= $SAPassW.GetNetworkCredential().Password
+
+    }
+    else {
+        & $SetupFile /ConfigurationFile=$configini /Q /IACCEPTSQLSERVERLICENSETERMS
+    }
+
+    
+
+    # Grant service account the right to perform volume maintenance
+    # code found at https://social.technet.microsoft.com/Forums/windows/en-US/5f293595-772e-4d0c-88af-f54e55814223/adding-domain-account-to-the-local-policy-user-rights-assignment-perform-volume-maintenance?forum=winserverpowershell
+
+    if ($PerformVolumeMaintenance) {
+        ## <--- Configure here
+        $accountToAdd = 'NT Service\MSSQL$AXIANSDB01'
+        ## ---> End of Config
+        $sidstr = $null
+
+
+        try {
+            $ntprincipal = new-object System.Security.Principal.NTAccount "$accountToAdd"
+            $sid = $ntprincipal.Translate([System.Security.Principal.SecurityIdentifier])
+            $sidstr = $sid.Value.ToString()
+        }
+        catch {
+            $sidstr = $null
+        }
+        Write-Message -Level Verbose -Message "Account: $($accountToAdd)" -ForegroundColor DarkCyan
+        if ( [string]::IsNullOrEmpty($sidstr) ) {
+            Write-Message -Level Verbose -Message "Account not found!" -ForegroundColor Red
+            #exit -1
+        }
+
+        Write-Message -Level Verbose -Message "Account SID: $($sidstr)" -ForegroundColor DarkCyan
```

Comment:
The colors are controlled by the config system, please remove the colors. At the end, we'll add some pizzaz with a progress bar if you'd like (I'd like!)

It would look something like this

![image](https://user-images.githubusercontent.com/8278033/47803745-76bd7b00-dd33-11e8-9a9b-bb787d26b084.png)

Reply:
Yeah that's really cool!